### PR TITLE
Feat/site editor

### PR DESCRIPTION
**Pull request by kaisermann**
**What problem is this solving?**

Would be sweet to be able to edit conditions via the site editor, this PR does that.

**How should this be manually tested?**

Go here: https://kiwi--storecomponents.myvtex.com/admin/cms/site-editor/traveler-backpack/p

**Screenshots or example usage:**

image

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -6,10 +6,11 @@ The condition layout app allows to conditionally render a block given certain co
 
 ## Configuration
 
-The app exports three kinds of block:
+The `condition-layout` app exports two kinds of block: `condition-layout.{context}` and `condition.{context}`.
 
 - `condition-layout.{context}` - These blocks are responsible for holding your condition blocks and providing the appropriate context.
 - `condition` - This blocks hold the condition logic and childrens to be displayed given the condition resolves to true.
+- `condition.{context}` - Same as `condition` but it provides a better editability via the Site Editor.
```
Comment (klzns) [inline docs/README.md:13]: Why one would choose `condition` over `condition.{context}`?

File: react/Condition.ts [hunk 2]
```diff
@@ -48,4 +57,59 @@ const Condition: FC<Props> = ({ children, conditions, match }) => {
   return (children as any) ?? null
 }
 
+Condition.schema = {
```
Comment (klzns) [inline react/Condition.ts:60]: Can we move this to `interfaces.json`? so we avoid loading this JSON in the storefront.

I think you should only have here a little schema with the title property so this blocks may appear in the site-editor

File: react/ConditionProduct.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+const ConditionProduct: StorefrontFunctionComponent<ConditionProps> = ({
+  children,
+  ...props
+}) => {
+  return <Condition {...props}>{children}</Condition>
+}
+
+ConditionProduct.schema = {
+  title: 'admin/editor.condition-layout.condition',
+  type: 'object',
```
Comment (klzns) [inline react/ConditionProduct.tsx:15]: same

File: docs/README.md (hunk 1 above)
Comment (kaisermann) [reply docs/README.md]: Done

File: docs/README.md [hunk 4]
```diff
@@ -6,10 +6,10 @@ The condition layout app allows to conditionally render a block given certain co
 
 ## Configuration
 
-The app exports three kinds of block:
+The `condition-layout` app exports two kinds of block: `condition-layout.{context}` and `condition.{context}`.
 
 - `condition-layout.{context}` - These blocks are responsible for holding your condition blocks and providing the appropriate context.
-- `condition` - This blocks hold the condition logic and childrens to be displayed given the condition resolves to true.
+- `condition.{context}` - These blocks hold the condition logic and childrens to be displayed given the condition resolves to true.
```
Comment (klzns) [inline docs/README.md:12]:
```suggestion
- `condition.{context}` - These blocks hold the condition logic and children to be displayed given the condition resolves to true.
```

File: store/interfaces.json [hunk 5]
```diff
@@ -2,16 +2,29 @@
   "condition-layout": {
     "component": "Noop"
   },
+  "condition-layout.product": {
+    "component": "ConditionLayoutProduct",
+    "composition": "children",
+    "content": {
+      "$ref": "#/definitions/ConditionLayoutProduct"
```
Comment (klzns) [inline store/interfaces.json:9]: Use full path so if people extend this it won't break

```suggestion
      "$ref": "app:vtex.condition-layout#/definitions/ConditionLayoutProduct"
```

File: store/interfaces.json [hunk 6]
```diff
@@ -2,16 +2,29 @@
...
     "component": "Condition",
     "composition": "children"
   },
   "condition.else": {
-    "component": "Else",
-    "composition": "children"
+    "component": "ConditionElse",
+    "composition": "children",
+    "content": {
+      "$ref": "#/definitions/ConditionElse"
```
Comment (klzns) [inline store/interfaces.json:20]:
```suggestion
      "$ref": "app:vtex.condition-layout#/definitions/ConditionElse"
```

File: store/interfaces.json [hunk 7]
```diff
@@ -2,16 +2,29 @@
...
+    }
   },
-  "condition-layout.product": {
-    "component": "Product",
-    "composition": "children"
+  "condition.product": {
+    "component": "ConditionProduct",
+    "composition": "children",
+    "content": {
+      "$ref": "#/definitions/ConditionProduct"
```
Comment (klzns) [inline store/interfaces.json:27]:
```suggestion
      "$ref": "app:vtex.condition-layout#/definitions/ConditionProduct"
```

**Pull request merged**